### PR TITLE
Karate documentation

### PR DESCRIPTION
**Pull request by simone-kalbermatter**
Adds and improves documentation for Karate

### CODE REVIEW

File: tests/karate/docs/README.md
Comment (pierluca) [inline tests/karate/docs/README.md:1]: If possible, put 1 sentence per line.
It makes diffs easier to read later on :-)

File: tests/karate/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+## Architecture
+### Features and Scenarios
+Karate test cases are called scenarios and they are grouped within different feature files. Each feature file tests a different message type (i.e. electionOpen, createRollCall etc.). 
+
+Scenarios are written in the Gerkhin syntax using the following keywords:
+
+- **Given**: Prepare the JSON payload to be sent to the component being tested. 
+- **When**: Defines the action that is to be performed with the payload. For instance, `publish` creates a message of type publish that contains some high-level message data, or `send` to send raw JSON data. 
+- **And**: Connector that can be used after any of the other keywords.
+- **Then**: Asserts that the action taken in the 'When' step has the expected outcome.
```
Comment (pierluca) [inline tests/karate/docs/README.md:17]: I'd present `Then` first, `And` later :)

File: tests/karate/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+- **Setting up previous steps necessary for a test**: For instance, before roll call messages can be tested, a LAO needs to be created first. `simpleScenarios.feature` contains many such useful setup steps.
+
+### Data model
+To generate valid message data for JSON payloads dynamically, a simplified version of the model is implemented in Java code. Mock components can create valid objects (for instance LAO, RollCall, Elections etc.), that can be used to handcraft messages. These objects also provide functions to override their own data to some invalid values, to craft invalid messages. 
+
+Some care needs to be taken if more of these functions to override valid data are implemented in the future. For example, when setting an invalid LAO name to test if the server rejects this, the LAO id also needs to be recomputed. The LAO id depends on the LAO name, and if the id is not recomputed the test might fail due to invalid LAO id, and not because the name was invalid. Currently, there is no way to distinguish this as only the error code is asserted and not the error message. This is a possible improvement that could be done in the future.
```
Comment (pierluca) [inline tests/karate/docs/README.md:29]: > Currently, there is no way to distinguish this as only the error code is asserted and not the error message

You can "future-proof" this by saying "As of February 2024, ...", which will make it obvious, later on, if this documentation needs to be taken with a pinch of salt or updated.

File: tests/karate/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+    * call read('classpath:be/features/utils/constants.feature')
+    * call read(serverFeature)
+    * call read(mockClientFeature)
+    * def organizer = call createMockFrontend
+    * def lao = organizer.createValidLao()
+    * def validRollCall = organizer.createValidRollCall(lao)
+    * call read(createLaoScenario) { organizer: '#(organizer)', lao: '#(lao)' }
+
+  @createRollCall1
```
Comment (pierluca) [inline tests/karate/docs/README.md:50]: You might want to explain why you put this tag here :)

File: tests/karate/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+## First Setup
+
+### All
+Make sure that you have [maven](https://maven.apache.org/download.cgi) installed.
+The Karate plugin for IntelliJ is also recommended.
+
+### Backend
+
+To test a backend, you don't need more setup that what is needed to build that backend. Use the resources provided by
+those projects.
```
Comment (pierluca) [inline tests/karate/docs/README.md:81]: here and below, the line breaks are a bit odd, and not aligned with sentences. You might want to adjust that :)

File: tests/karate/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+### Web Front-end
+
+Make sure you have [Google Chrome](https://www.google.com/intl/en/chrome/) and [npm](https://nodejs.org/en/download/)
+installed.
+
+## Running the Tests
+
+### Backend
+
+Build the backend you want to test. Follow the steps described in the corresponding subproject.
```
Comment (pierluca) [inline tests/karate/docs/README.md:109]: How does Karate find the built executables ? Does it need to be copied over ? 🤔 
I don't remember this part.

File: tests/karate/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+Start the Android Emulator. Start Appium : if you use the GUI, delete the text in Host and Port and click on the start
+server button. If you use the terminal, run `appium`.
+
+With Android Bumblebee the emulator can either run in a tool window or a standalone window. (To have it in a standalone
+window, go to `File -> Settings -> Tools -> Emulator` and unselect `Launch in a tool window`).
+- Standalone window : \
+The emulator window name should match : `Android Emulator - avd:id` \
+Ex: `Android Emulator - Pixel_4_API_30:5554`
+- Tool window : open the `Extended Controls` (the 3 points above the emulator)
```
Comment (pierluca) [inline tests/karate/docs/README.md:139]: > the 3 points above the emulator

This kind of thing might be clearer with a couple of screenshots.

**Review by pierluca — CHANGES_REQUESTED** [review]
Pretty good, minor points :-)

File: tests/karate/docs/README.md (hunk 4 above)
Comment (simone-kalbermatter) [reply tests/karate/docs/README.md]: I copied this from the old documentation and didn't see this, thanks 👍

File: tests/karate/docs/README.md (hunk 6 above)
Comment (simone-kalbermatter) [reply tests/karate/docs/README.md:164]: I don't have access to Android Studio right now, but I can do this at the end of the week.

**Review by pierluca — APPROVED** [review]

**Pull request merged**